src/mesh/mesh.cc

```diff
@@ -889,31 +889,32 @@ bool compare_points(const arma::vec3 &p1, const arma::vec3 &p2) {
 }
 
 
-bool Mesh::check_compatible_mesh( Mesh & mesh, vector<LongIdx> & bulk_elements_id, vector<LongIdx> & boundary_elements_id )
+bool Mesh::check_compatible_mesh( Mesh & computational_mesh, vector<LongIdx> & element_ids_map )
 {
 	std::vector<unsigned int> node_ids; // allow mapping ids of nodes from source mesh to target mesh
 	std::vector<unsigned int> node_list;
 	std::vector<unsigned int> candidate_list; // returned by intersect_element_lists
 	std::vector<unsigned int> result_list; // list of elements with same dimension as vtk element
 	unsigned int i; // counter over vectors
+	element_ids_map.resize(computational_mesh.n_elements()+computational_mesh.n_elements(true));
 
     {
         // iterates over node vector of \p this object
         // to each node must be found just only one node in target \p mesh
         // store orders (mapping between source and target meshes) into node_ids vector
         std::vector<unsigned int> searched_elements; // for BIH tree
         unsigned int i_node, i_elm_node;
-        const BIHTree &bih_tree=mesh.get_bih_tree();
+        const BIHTree &bih_tree=this->get_bih_tree();
 
     	// create nodes of mesh
-        node_ids.resize( this->n_nodes() );
+        node_ids.resize( computational_mesh.n_nodes() );
```

Comment:
E.g. here just initialize to undef_idx and then pass through the input mesh, tr to find the node in the computation mesh and possibly mark the input node index in to the map.